js/src/lib/utils/convert.ts

```diff
@@ -34,7 +34,16 @@ export const bytesTA = function (arg: string) {
 }
 
 export const numberTB = function (arg: BN) {
-  return arg.toNumber();
+  let res: BN | number;
+  try {
+    // number is limited to 53 bits, BN will throw Error
+    res = arg.toNumber();
+  }
+  catch {
```

Comment:
Any chance we could be more specific about the catch here? If it would rely on something fragile like matching a string then don't bother.

Reply:
Yes, I had considered that but bn.js simply does a `throw Error(''Number can only safely store up to 53 bits')` 
see https://github.com/indutny/bn.js/blob/master/lib/bn.js#L547  and https://github.com/indutny/bn.js/blob/master/lib/bn.js#L5